addons/auth_signup/models/res_partner.py

```diff
@@ -57,47 +57,51 @@ def _get_signup_url_for_action(self, url=None, action=None, view_type=None, menu
-            if signup_type:
-                route = 'reset_password' if signup_type == 'reset' else signup_type
-
-            if partner.sudo().signup_token and signup_type:
-                query['token'] = partner.sudo().signup_token
-            elif partner.user_ids:
-                query['login'] = partner.user_ids[0].login
+            # the parameters to encode for the query
+            if signup_type == 'reset':
+                query = {'db': self.env.cr.dbname}
             else:
-                continue        # no signup token, no user, thus no signup url!
+                query = {'db': self.env.cr.dbname, 'signup_email': partner.email}
+        if signup_type:
+            route = 'reset_password' if signup_type == 'reset' else signup_type
+
+        if partner.sudo().signup_token and signup_type:
+            query['token'] = partner.sudo().signup_token
+        elif partner.user_ids:
+            query['login'] = partner.user_ids[0].login
+        else:
+            continue  # no signup token, no user, thus no signup url!
 
-            if url:
-                query['redirect'] = url
-            else:
-                fragment = dict()
-                base = '/web#'
-                if action == '/mail/view':
-                    base = '/mail/view?'
-                elif action:
-                    fragment['action'] = action
-                if view_type:
-                    fragment['view_type'] = view_type
-                if menu_id:
-                    fragment['menu_id'] = menu_id
-                if model:
-                    fragment['model'] = model
-                if res_id:
-                    fragment['res_id'] = res_id
-
-                if fragment:
-                    query['redirect'] = base + werkzeug.urls.url_encode(fragment)
-
-            signup_url = "/web/%s?%s" % (route, werkzeug.urls.url_encode(query))
-            if not self.env.context.get('relative_url'):
-                signup_url = werkzeug.urls.url_join(base_url, signup_url)
-            res[partner.id] = signup_url
+        if url:
+            query['redirect'] = url
+        else:
+            fragment = dict()
+            base = '/web#'
+            if action == '/mail/view':
+                base = '/mail/view?'
+            elif action:
+                fragment['action'] = action
+            if view_type:
+                fragment['view_type'] = view_type
+            if menu_id:
+                fragment['menu_id'] = menu_id
+            if model:
+                fragment['model'] = model
+            if res_id:
+                fragment['res_id'] = res_id
+
+            if fragment:
+                query['redirect'] = base + werkzeug.urls.url_encode(fragment)
+
+        signup_url = "/web/%s?%s" % (route, werkzeug.urls.url_encode(query))
+        if not self.env.context.get('relative_url'):
+            signup_url = werkzeug.urls.url_join(base_url, signup_url)
+        res[partner.id] = signup_url
```

Comment:
You must indent ones to the right of all this code selected. The code selected must be inside the `for partner in self:`